### PR TITLE
monit: add support for all monit services when checking process state

### PR DESCRIPTION
**Pull request by gherceg**
##### SUMMARY

The changes introduced by @snopoke [here](https://github.com/ansible-collections/community.general/pull/1107) did not include support for other [monit service checks](https://mmonit.com/monit/documentation/monit.html#Service-checks) besides "Process". Other service checks return a process status of "missing".

This change checks the output of `monit validate <name>` or `monit status <name>` for all of the potential monit service checks. I've tested this locally when running the "File" service check and it works as expected. The "File" service check is also how I noticed the issue in the first place. 

The unit test assumes the text returned from other service checks based on the pattern observed in the "Process" and "File" service checks. To be safe, this was my reasoning for ignoring case in the check (specifically for the "Remote host" service).

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

monit

### CODE REVIEW

File: plugins/modules/monitoring/monit.py [hunk 1]
```diff
@@ -151,7 +154,8 @@ def get_status(self, validate=False):
         return self._parse_status(out, err)
 
     def _parse_status(self, output, err):
-        if "Process '%s'" % self.process_name not in output:
+        pattern = "(%s) '%s'" % ('|'.join(MONIT_SERVICES), self.process_name)
```
Comment (felixfontein) [inline plugins/modules/monitoring/monit.py:157]:
```suggestion
        pattern = "(%s) '%s'" % ('|'.join(re.escape(MONIT_SERVICES)), re.escape(self.process_name))
```

**Review by felixfontein — COMMENTED** [review]
You need to add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

**Comment by gherceg** [comment]
Ah must have misread the changelog policy. Sorry about the messy commit history. Thanks.

**Review by felixfontein — APPROVED** [review]
Change looks good to me, but I don't know and use monit. Dear maintainers, please take a look at whether this change is semantically correct.

**Comment by felixfontein** [comment]
Restart CI

**Comment by felixfontein** [comment]
@gherceg thanks for fixing this!

**Review by snopoke — COMMENTED** [review]
Thanks for the changes, they look fine to me. Apologies for the delayed response (I've been away over the holidays).

**Pull request merged**